lib/modifiers/flatten.js

```diff
@@ -0,0 +1,60 @@
+var _ = require('lodash');
+var through = require('through2');
+
+/**
+ * Returns a new stream that flattens all arrays passing through by one level. All non-array
+ * items will be passed through as-is.
+ *
+ * @method
+ * @static
+ * @since    1.5.0
+ * @category Util
+ *
+ * @returns {Stream} - Transform stream.
+ *
+ * @example
+ * var input = through.obj();
+ *
+ * input.pipe(nodeStream.flatten());
+ *
+ * input.write([1, 2, 3]);
+ * input.write([4, 5]);
+ * input.write(6);
+ *
+ * // => [1, 2, 3, 4, 5, 6]
+ * @example
+ * var input = through.obj();
+ *
+ * input
+ *   // batch items that are read
+ *   .pipe(nodeStream.batch({ count: 2 }))
+ *   // perform a transform action on the batches
+ *   .pipe(transformBatch())
+ *   // flatten the batches back
+ *   .pipe(nodeStream.flatten());
+ *
+ * input.write(1);
+ * input.write(2);
+ * input.write(3);
+ * input.write(4);
+ * input.write(5);
+ *
+ * // => [1, 2, 3, 4, 5]
+ */
+function flatten() {
+  return through.obj(function Transform(data, enc, cb) {
+    var self = this;
+
+    if (!_.isArray(data)) {
+      return cb(null, data);
+    }
+
+    data.forEach(function (val) {
+      self.push(val);
```

Comment:
More of a question than anything else: Would it make sense to add these to the stream asynchronously? The idea being if we make a big enough array we'll fill up the buffer (16 items) without giving consumers any chance to read it. If we push to the stream asynchronously, we can allow consumers to read between writes and the stream will flow more steadily.